packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -514,21 +548,33 @@ export type CustomSelectProps<T> = SelectCommonProps & {
    * Props to pass to the select button (trigger) element.
    */
   triggerProps?: React.ComponentPropsWithRef<'div'>;
-} & SelectMultipleTypeProps<T> &
+} & CustomSelectMultipleAndStyleTypeProps<T> &
   Omit<
     React.ComponentPropsWithoutRef<'div'>,
-    'size' | 'disabled' | 'placeholder' | 'onChange'
+    'size' | 'disabled' | 'placeholder' | 'onChange' | 'defaultValue'
   >;
 
 export type SelectValueChangeEvent = 'added' | 'removed';
 
-export type SelectMultipleTypeProps<T> =
+export type CustomSelectMultipleAndStyleTypeProps<T> =
   | {
       /**
        * Enable multiple selection.
        * @default false
        */
       multiple?: false;
+      /**
+       * Style of the select.
+       * Use 'borderless' to hide outline.
+       * @default 'default'
+       */
+      styleType?: 'default';
+      /**
+       * Placeholder when no item is selected.
+       *
+       * When `styleType=borderless`, `placeholder` is not allowed. Additionally, a `defaultValue` must be provided.
+       */
+      placeholder?: React.ReactNode;
```

Comment:
nit: I think the jsdocs are now repeated and could be removed in certain places?

Reply:
Yes, maybe. Actually, I was hoping we address it later along with https://github.com/iTwin/iTwinUI/pull/1862#discussion_r1492672122. That's why I didn't test the JSDocs too much and just concentrated on the logic.